src/jit/importer.cpp

```diff
@@ -19601,14 +19593,7 @@ void Compiler::impDevirtualizeCall(GenTreeCall*            call,
         // the time of jitting, objClass has no subclasses that
         // override this method), then perhaps we'd be willing to
         // make a bet...?
-        JITDUMP("    Class not final or exact, method not final, no devirtualization\n");
-        return;
```

Comment:
It might be worth adding a guard check and doing devirtualization for the methods that are marked as final.
For now you can update the comment.   Thought we can't "trust" final, it is in fact a pretty valuable hint.